src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -277,38 +305,24 @@ private void doGetIntern(HttpServletRequest req, HttpServletResponse resp)
 
 			StringBuilder awsSessionName = new StringBuilder();
 			boolean first=true;
-			for (String claimName : getClaimNames()) {
-				String claimValue = jwt.getBody().get(claimName, String.class);
+			for (String claimName : getSessionClaimNames()) {
+				String claimValue = claims.get(claimName, String.class);
 				if (StringUtils.isEmpty(claimValue)) continue;
 				if (first) first=false; else awsSessionName.append(":");
 				awsSessionName.append(claimValue);
 			}
 
 			// get STS token
-			AssumeRoleWithWebIdentityRequest assumeRoleWithWebIdentityRequest = new AssumeRoleWithWebIdentityRequest();
-			assumeRoleWithWebIdentityRequest.setWebIdentityToken(idToken.getToken());
-			assumeRoleWithWebIdentityRequest.setRoleArn(roleArn);
-			assumeRoleWithWebIdentityRequest.setRoleSessionName(awsSessionName.toString());
+			AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest();
+			assumeRoleRequest.setRoleArn(roleArn);
+			assumeRoleRequest.setRoleSessionName(awsSessionName.toString());
 			AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder.standard()
-					.withRegion(Regions.fromName(awsRegion))
-					.withCredentials(new AWSCredentialsProvider() {
-						@Override
-						public AWSCredentials getCredentials() {
-							return new AWSCredentials() {
-								@Override
-								public String getAWSAccessKeyId() {
-									return "dummyKeyId";
-								}
-								@Override
-								public String getAWSSecretKey() {
-									return "dummySecret";
-								}};
-						}
-						@Override
-						public void refresh() {}}).build();
-			
-			AssumeRoleWithWebIdentityResult assumeRoleWithWebIdentityResult = stsClient.assumeRoleWithWebIdentity(assumeRoleWithWebIdentityRequest);
-			Credentials credentials = assumeRoleWithWebIdentityResult.getCredentials();
+					.withRegion(Regions.fromName(awsRegion)).build();
+			Collection<Tag> tags = new ArrayList<Tag>();
+			tags.add(new Tag().withKey("foo").withValue("bar"));
```

Comment:
what's up with this?  the client just needs some dummy tags?

Reply:
It's a bug.  I factored out the code into a testable method and added the needed test.